### PR TITLE
MINIFICPP-2401 Make RocksDB verify_checksums read option configurable

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-2401

---------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by szaszm** [comment]
If it's not too much additional effort, I think it would be better to make this configurable per-repository. I agree with the changed default to false.

**Comment by lordgamez** [comment]
> If it's not too much additional effort, I think it would be better to make this configurable per-repository. I agree with the changed default to false, but I think the PR title and/or commit message should mention this important change.

Good idea, updated in 38d369caf2fc1c199a440bb6e368978d4c97e98a

File: extensions/rocksdb-repos/FlowFileRepository.cpp [hunk 1]
```diff
@@ -142,7 +144,9 @@ void FlowFileRepository::initialize_repository() {
   }
   logger_->log_info("Reading existing flow files from database");
 
-  const auto it = opendb->NewIterator(rocksdb::ReadOptions());
+  rocksdb::ReadOptions options;
+  options.verify_checksums = verify_checksums_in_rocksdb_reads_;
+  const auto it = opendb->NewIterator(options);
```
Comment (szaszm) [inline extensions/rocksdb-repos/FlowFileRepository.cpp:149]:
```suggestion
  const auto it = opendb->NewIterator(rocksdb::ReadOptions{
    .verify_checksums = verify_checksums_in_rocksdb_reads_
  });
```

File: extensions/rocksdb-repos/FlowFileRepository.cpp [hunk 2]
```diff
@@ -208,6 +212,9 @@ bool FlowFileRepository::initialize(const std::shared_ptr<Configure> &configure)
   const auto encrypted_env = createEncryptingEnv(utils::crypto::EncryptionManager{configure->getHome()}, DbEncryptionOptions{directory_, ENCRYPTION_KEY_NAME});
   logger_->log_info("Using {} FlowFileRepository", encrypted_env ? "encrypted" : "plaintext");
 
+  verify_checksums_in_rocksdb_reads_ = configure->get(Configure::nifi_flowfile_repository_rocksdb_read_verify_checksums).value_or("false") == "true";
```
Comment (szaszm) [inline extensions/rocksdb-repos/FlowFileRepository.cpp:215]:
```suggestion
  verify_checksums_in_rocksdb_reads_ = (configure->get(Configure::nifi_flowfile_repository_rocksdb_read_verify_checksums) | utils::flatMap(&utils::string::toBool)).value_or("false");
```

**Review by adamdebreceni — APPROVED** [review]

File: extensions/rocksdb-repos/FlowFileRepository.cpp (hunk 1 above)
Comment (lordgamez) [reply extensions/rocksdb-repos/FlowFileRepository.cpp:149]: Unfortunately it does not work with this type:
```
error: designated initializers cannot be used with a non-aggregate type ‘rocksdb::ReadOptions’
```

File: extensions/rocksdb-repos/FlowFileRepository.cpp (hunk 2 above)
Comment (lordgamez) [reply extensions/rocksdb-repos/FlowFileRepository.cpp]: Updated in 349797104ca89b54d1fce4e260d12d52737876b4

**Review by szaszm — APPROVED** [review]

**Pull request closed without merging**